ide/che-core-ide-stacks/src/main/resources/stacks.json

```diff
@@ -2561,5 +2561,67 @@
       "name": "type-apache-camel.svg",
       "mediaType": "image/svg+xml"
     }
+  },
+  {
+    "id": "che7-python",
+    "name": "Che 7 Python",
+    "creator": "Anatolii Bazko",
+    "tags": [
+      "ws.next",
+      "Theia",
+      "Python",
+      "pip"
+    ],
+    "scope": "general",
+    "description": "Workspace.next sidecars and Theia as IDE",
+    "components": [
+      {
+        "version": "7",
+        "name": "Centos"
+      },
+      {
+        "version": "3.7.2",
+        "name": "Python"
+      },
+      {
+        "version": "19.0.3 ",
+        "name": "PIP"
+      }
+    ],
+    "workspaceConfig": {
+      "environments": {
+        "default": {
+          "recipe": {
+            "contentType": "application/x-yaml",
+            "type": "kubernetes",
+            "content": "kind: List\nitems:\n - \n  apiVersion: v1\n  kind: Pod\n  metadata:\n   name: ws\n  spec:\n   containers:\n    - \n     image: 'eclipse/che-dev:nightly'\n     name: dev\n     resources:\n      limits:\n       memory: 512Mi\n"
+          },
+          "machines": {
+            "ws/dev": {
+              "env": {},
+              "servers": {},
+              "volumes": {
+                "projects": {
+                  "path": "/projects"
+                }
+              },
+              "installers": [],
+              "attributes": {
+                "memoryLimitBytes": "536870912"
```

Comment:
Please define memory in one place, recipe or machine attributes. Defining it in both places at the same time does not make sense and may be confusing.
Note that in this case, memory limit from attributes will win and override limit in recipe content.